src/DotNetCore.CAP.SqlServer/Diagnostics/DiagnosticObserver.cs

```diff
@@ -50,13 +50,19 @@ public void OnNext(KeyValuePair<string, object> evt)
             }
             else if (evt.Key == SqlErrorCommitTransaction || evt.Key == SqlErrorCommitTransactionMicrosoft)
             {
-                var sqlConnection = (SqlConnection)GetProperty(evt.Value, "Connection");
+                if (!GetSqlConnection(evt, out SqlConnection sqlConnection)) return;
                 var transactionKey = sqlConnection.ClientConnectionId;
 
                 _bufferList.TryRemove(transactionKey, out _);
             }
         }
 
+        private bool GetSqlConnection(KeyValuePair<string, object> evt, out SqlConnection sqlConnection)
```

Comment:
I think to use `TryGetSqlConnection` as the method name and  defined as the static method will make more sense

Reply:
Yes, `TryGetSqlConnection` fits better with Microsoft APIs. I pushed this change as well. Thanks.